gson/src/test/java/com/google/gson/JsonStreamParserTest.java

```diff
@@ -72,57 +72,33 @@ public void testNoSideEffectForHasNext() {
   public void testCallingNextBeyondAvailableInput() {
     JsonElement unused1 = parser.next();
     JsonElement unused2 = parser.next();
-    try {
-      parser.next();
-      fail("Parser should not go beyond available input");
-    } catch (NoSuchElementException expected) {
-    }
+    assertThrows(NoSuchElementException.class, parser::next);
```

Comment:
Maybe add the previous message as comment so that it is clear why this should fail:
```suggestion
    // Parser should not go beyond available input
    assertThrows(NoSuchElementException.class, parser::next);
```